### ISSUE TITLE
"Internal Server Error" when connecting via Github Actions

### ISSUE THREAD

**Issue by twfahey1** [zerotier/github-action#15]
We're seeing a consistent error the last 24 hours with our Github Action ZeroTier connection:
```
⏁  Installing ZeroTier
⏁  Authorizing Runner to ZeroTier network
Internal Server Error
⏁  Joining ZeroTier Network ID: ***
200 join OK
```
At this point the step just hangs until timeout. Is this something we can address on our side somehow? We aren't detecting any other problems with our ZeroTier connection apart from our Action, which we've been using for a long time now. Any feedback / ideas appreciated.

**Comment by dfiru** [comment]
same.

**Comment by twfahey1** [comment]
Update: I made a fork to do debugging - https://github.com/twfahey1/github-action - Oddly, with only echoing out and debugging parameters in place... it succeeded... Confirming it wasn't a fluke...

**Comment by twfahey1** [comment]
Ha, interestingly... That seems to be working now! Will update if anything changes, but, I just wired up my fork across our suite of jobs that implement several back-to-back ZT connections, and it went off as normal.

I noticed that in the output with my debugging, I'm now seeing an initial failure, then success...
```
⏁  Authorizing Runner to ZeroTier network
DEBUG: API URL: https://my.zerotier.com/api/v1
DEBUG: Network ID: ***
DEBUG: Member ID: [the right data]
DEBUG: Auth Token: ***
DEBUG: Constructed URL: https://my.zerotier.com/api/v1/network/***/member/****
DEBUG: GitHub Member: Zerotier GitHub Member 'the right data here'
DEBUG: GitHub Server URL: https://github.com/
DEBUG: GitHub Repo: The right repo
DEBUG: GitHub Run ID: The right num
Authorization failed. Retrying in 2 seconds... (Attempt 1 of 10)
{All expected JSON appears now}
Member authorized successfully.
```
 I'm at this point assuming something is happening along the lines of the initial authentication is failing, and perhaps the Linux environment is such that the retry logic is not working as expected... and somehow by echoing things out, we're working around this error? 

When I have more time, I could test out a simple echo of "Initiating..." or something, which might be more mergeable if a better fix isn't available....

**Comment by twfahey1** [comment]
To be clear, I replaced instances of `zerotier/github-action@v1` with my fork and branch I made the updates on: `twfahey1/github-action@main`

**Comment by orensbruli** [comment]
I made a PR recently and now they have a 1.0.3 version of the action.
@twfahey1 would you mind testing it?
I also see that there's an initial Authorization failed but I don't know why.

Also, maybe it would be useful if you could make a PR with your debug strings with the conditional ${{ runner.debug }} variable?
So if the action workflow is run in debug mode we can check what is happening?

**Comment by twfahey1** [comment]
Thanks @orensbruli ! I realize now that I was using v1... so with your implementation in main, I was seeing the fix in my fork! It looks like using `v1.0.3` all is well. I have moved our implementation back to this explicit version

<img width="250" alt="image" src="https://github.com/zerotier/github-action/assets/6699959/f337b89d-497c-44e4-ba91-401d2b1a7299">
